### DISCUSSION TITLE
Problem when connecting

### DISCUSSION THREAD

**Discussion by Siber0817** [itzg/docker-minecraft-server#1706] [Q&A] [answered]
Hi.
I just installed this on my personal server but i found out that i cannot connect using a cracked mc launcher. It there a way to disable this security feature like it was possible, when i used to run a minecraft server from windows? I believe it was in config but i cannot find the directory of my mc server in my debian 11 machine.
Sorry if it sound weird but i'm not english so i hope that what i meant is understandable.
Thanks

**Comment by Siber0817** [comment, marked as answer]
Ok i just found it. Literally seconds after i posted this i found the directory. I must be blind. Sorry for bothering.

**Reply by Avergaminecraftplayer** [reply]
By any chance could you tell me which directory it is and what did you do? Im having trouble getting my brother in the server and im really new at linux

**Reply by Siber0817** [reply]
Hi,
So i found out that i had to set a directory. In my initial docker file a didnt set it so it defaulted to some place that i dont know. What you want to do is for using docker run command, use -v /var/data:/data , which would set the /var/data directory as the directory for your server. If youre using docker compose, add the lines volumes:
-./var/data:/data

One downside is it will rebuild your server, so new world etc but you probably could back it up somehow and then replace the files. After you have the directory up and running, you can change the server.properties file to allow for cracker players.
Hope this helps you

**Reply by Siber0817** [reply]
Oh and the /var/data was a example, you can use whatever directory you want

**Reply by itzg** [reply]
FYI more info on the data directory, converting anonymous volumes, etc

https://docker-minecraft-server.readthedocs.io/en/latest/data-directory/

**Reply by Avergaminecraftplayer** [reply]
> Oh and the /var/data was a example, you can use whatever directory you want

Hello thanks for the response, turns out i had the directory right, what i was trying to ask is how did you manage to get the cracked minecraft running, i already tried to add the ONLINE_MODE: "FALSE" var in the enviroment section of the code, but that just makes it so no one can join, be it premium or cracked users

**Reply by Siber0817** [reply]
Hi. So to do that, you'll have to open server.properties and change "online-mod=true" to "online-mod=false". Then just save and restart your server. You dont need to add anything, just change this one line

**Reply by Avergaminecraftplayer** [reply]
> Hi. So to do that, you'll have to open server.properties and change "online-mod=true" to "online-mod=false". Then just save and restart your server. You dont need to add anything, just change this one line

Hey so, this thing is already at false... tried to delete the directive i made in the environment section of the code, but it still doesnt let neither the premium and the cracked version join

**Reply by Siber0817** [reply]
That should be the only thing blocking cracked players. Also its weird that premium wont join. When you add the server to your server list, does it write something or does it write cant connect? And when you connect to the server, what does it write? Ill try to diagnose this for you but i believe you have a problem with your networking, as of static IP of server or something like that.

**Reply by Siber0817** [reply]
Also, before you added that directive, did it worked for premium? Something seems wrong but i cant just caught on to what happened yet

**Reply by Avergaminecraftplayer** [reply]
> That should be the only thing blocking cracked players. Also its weird that premium wont join. When you add the server to your server list, does it write something or does it write cant connect? And when you connect to the server, what does it write? Ill try to diagnose this for you but i believe you have a problem with your networking, as of static IP of server or something like that.

So, premium CAN join with the online mode active, but cracked cant, and the server is op and running in the server list, the MOTD works fine, and the connection state has the four green lines, when both premium and cracked players try to connect an "You're not in the server white list!" message appears

**Reply by Siber0817** [reply]
Well you should have written that sooner xD. The problem is you somehow managed to activate the whitelist function of MC server while allowing for cracked players. I dont know how, but i dont think that it matters too much. What you need to do is disable whitelist. There are a few ways of doing that. First is while your server is running, go to console and write /whitelist off. Then restart. The other option is going to server.properties and finding a line "enforce-whitelist=TRUE" and change it to "FALSE". Then restart. I recommend going the console route. Let me know if it helps

**Reply by Siber0817** [reply]
Oh and also there is a line "whitelist=TRUE" so change that to "FALSE" too. I just noticed that there are 2 lines about it so make sure everything that has "whitelist" in it is set to "FALSE"

**Reply by Avergaminecraftplayer** [reply]
> Oh and also there is a line "whitelist=TRUE" so change that to "FALSE" too. I just noticed that there are 2 lines about it so make sure everything that has "whitelist" in it is set to "FALSE"

Worked like a charm dude, many thanks

**Reply by Siber0817** [reply]
> > Oh and also there is a line "whitelist=TRUE" so change that to "FALSE" too. I just noticed that there are 2 lines about it so make sure everything that has "whitelist" in it is set to "FALSE"
> 
> Worked like a charm dude, many thanks

No problem, glad that i could help
